src/ERC1155A.sol

```diff
@@ -554,23 +555,28 @@ abstract contract ERC1155A is IERC1155A {
             }
         }
 
-        emit TransferBatch(msg.sender, from, address(0), ids, amounts);
+        emit TransferBatch(operator, from, address(0), ids, amounts);
     }
 
-    /// @dev Implementation copied from solmate/ERC1155
-    function _burn(address from, uint256 id, uint256 amount) internal virtual {
+    /// @dev Implementation copied from solmate/ERC1155 and adapted with operator logic
+    function _burn(address from, address operator, uint256 id, uint256 amount) internal virtual {
         // Check if the msg.sender is the owner or is approved for all tokens
-        if (msg.sender != from && !isApprovedForAll[from][msg.sender]) {
+        if (operator != from && !isApprovedForAll[from][operator]) {
             // If not, then check if the msg.sender has sufficient allowance
-            require(allowance(from, msg.sender, id) >= amount, "NOT_AUTHORIZED");
-            allowances[from][msg.sender][id] -= amount; // Deduct the burned amount from the allowance
+            require(allowance(from, operator, id) >= amount, "NOT_AUTHORIZED");
```

Comment:
string based error message (could be custom error)

Reply:
fixed